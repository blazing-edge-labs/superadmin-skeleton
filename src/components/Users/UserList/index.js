import React from 'react'
import {
  List,
  Datagrid,
  TextField,
  EmailField,
  DateField,
  EditButton,
  DeleteButton,
  Filter,
  TextInput,
} from 'admin-on-rest'

const UserFilter = (props) => (
  <Filter {...props}>
    <TextInput label="Email" source="email"/>
  </Filter>
)

const UserList = (props) => (
  <List
    {...props}
    perPage={10}
    sort={{ field: 'id', order: 'ASC' }}
    filters={<UserFilter/>}
  >
    <Datagrid>
      <TextField source="id"/>
      <EmailField source="email"/>
      <DateField source="createdAt" label="Created At" showTime options={{ timeZoneName: 'short' }}/>
      <EditButton/>
      <DeleteButton/>
    </Datagrid>
  </List>
)

export default UserList
