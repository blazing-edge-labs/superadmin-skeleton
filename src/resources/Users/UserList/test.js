import React from 'react'
import { shallow } from 'enzyme'

import UserList from './'

describe('UserList', () => {
  const userList = shallow(<UserList/>)

  it('renders properly', () => {
    expect(userList).toMatchSnapshot()
  })
})
