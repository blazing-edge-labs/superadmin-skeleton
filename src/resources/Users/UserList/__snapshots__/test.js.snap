// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders properly 1`] = `
<Connect(getContext(List))
  filters={<UserFilter />}
  perPage={10}
  sort={
    Object {
      "field": "id",
      "order": "ASC",
    }
  }
>
  <MuiComponent>
    <pure(TextField)
      addLabel={true}
      source="id"
    />
    <pure(EmailField)
      addLabel={true}
      source="email"
    />
    <pure(DateField)
      addLabel={true}
      label="Created At"
      options={
        Object {
          "timeZoneName": "short",
        }
      }
      showTime={true}
      source="createdAt"
    />
    <shouldUpdate(getContext(EditButton)) />
    <getContext(DeleteButton) />
  </MuiComponent>
</Connect(getContext(List))>
`;
